mithril-aggregator/src/metrics/service.rs

```diff
@@ -146,20 +147,59 @@ impl MetricsService {
             .join(",")
     }
 
-    fn build_metric_map_per_label(&self, metric_family: &MetricFamily) -> HashMap<String, u32> {
+    fn build_metric_map(
+        &self,
+        metric_family: &MetricFamily,
+    ) -> HashMap<String, MetricLabelValueMap> {
         metric_family
             .get_metric()
             .iter()
             .map(|m| {
                 (
                     self.build_label_key(m.get_label()),
-                    m.get_counter().as_ref().unwrap_or_default().value() as u32,
+                    MetricLabelValueMap::new(m.clone()),
                 )
             })
             .collect()
     }
 }
 
+type LabelName = String;
+type LabelValue = String;
+#[derive(Debug, Clone, PartialEq, Eq)]
+
+/// MetricLabelValueMap represent a metric with a LabelName/LabelValue map and a counter.
```

Comment:
I recommend using the types with brackets in the documentation. That way, if you ever rename it, you won’t risk forgetting to update the doc comment.
```suggestion
/// [MetricLabelValueMap] represent a metric with a [LabelName]/[LabelValue] map and a counter.
```